src/WebJobs.Script/Host/WorkerFunctionMetadataProvider.cs

```diff
@@ -65,7 +65,7 @@ public async Task<FunctionMetadataResult> GetFunctionMetadataAsync(IEnumerable<R
 
                 if (channels?.Any() != true)
                 {
-                    await _channelManager.InitializeChannelAsync(_workerRuntime);
+                    await _channelManager.InitializeChannelAsync(workerConfigs, _workerRuntime);
```

Comment:
Here, we've been given the correct workerConfigs to use, but they were never being used for anything. This change now uses them for initialization, which allows a JobHost to override the WebHost, which is a requirement for dotnet-isolated.